src/shims/unix/solarish/foreign_items.rs

```diff
@@ -57,6 +57,32 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 this.write_scalar(res, dest)?;
             }
 
+            // File related shims
+            "stat" | "stat64" => {
+                let [path, buf] =
+                    this.check_shim(abi, ExternAbi::C { unwind: false }, link_name, args)?;
+                let result = this.macos_fbsd_solaris_stat(path, buf)?;
+                this.write_scalar(result, dest)?;
+            }
+            "lstat" | "lstat64" => {
+                let [path, buf] =
+                    this.check_shim(abi, ExternAbi::C { unwind: false }, link_name, args)?;
+                let result = this.macos_fbsd_solaris_lstat(path, buf)?;
+                this.write_scalar(result, dest)?;
+            }
+            "fstat" | "fstat64" => {
+                let [fd, buf] =
+                    this.check_shim(abi, ExternAbi::C { unwind: false }, link_name, args)?;
+                let result = this.macos_fbsd_solaris_fstat(fd, buf)?;
+                this.write_scalar(result, dest)?;
+            }
+            "readdir_r" => {
+                let [dirp, entry, result] =
+                    this.check_shim(abi, ExternAbi::C { unwind: false }, link_name, args)?;
+                let result = this.macos_fbsd_solaris_readdir_r(dirp, entry, result)?;
+                this.write_scalar(result, dest)?;
+            }
```

Comment:
I am confused -- why is readdir implemented now?

Please don't add untested code. If it "just works" without any other changes, enable the test, otherwise remove the implementation.

Reply:
I did not add readdir**_r** it had been there since [in the early phase](https://github.com/rust-lang/miri/commit/13b741d1c858405b85c94510dec678b4571f52a0). Anyhow, I do not need it at all for now.